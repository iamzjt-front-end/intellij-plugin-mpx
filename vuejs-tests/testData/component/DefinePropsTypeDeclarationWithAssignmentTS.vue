<script setup lang="ts">
const props = defineProps<{
  optional?: string,
  required: string,
}>();

props.optional;
<error descr="Unresolved variable or type optional">optional</error>;
</script>

<template>
  {{optional}}
  {{required}}
  {{props.optional}}
</template>
