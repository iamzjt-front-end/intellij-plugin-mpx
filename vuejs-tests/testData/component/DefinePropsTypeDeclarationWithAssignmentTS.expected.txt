{
 class: "VueSourceComponent",
 components: {
  DefinePropsTypeDeclarationWithAssignmentTS: <recursive value of class VueSourceComponent>,
 },
 props: [
  {
   class: "VueScriptSetupInputProperty",
   name: "optional",
   jsType: "string",
   required: false,
  },
  {
   class: "VueScriptSetupInputProperty",
   name: "required",
   jsType: "string",
   required: true,
  },
 ],
}
